libs/x-content/src/main/java/org/elasticsearch/common/xcontent/support/AbstractXContentParser.java

```diff
@@ -386,26 +391,20 @@ public CharBuffer charBufferOrNull() throws IOException {
 
         ArrayList<Object> list = new ArrayList<>();
         for (; token != null && token != XContentParser.Token.END_ARRAY; token = parser.nextToken()) {
-            list.add(readValue(parser, mapFactory, token));
+            list.add(readValue(parser, mapFactory));
         }
         return list;
     }
 
-    static Object readValue(XContentParser parser, MapFactory mapFactory, XContentParser.Token token) throws IOException {
-        if (token == XContentParser.Token.VALUE_NULL) {
-            return null;
-        } else if (token == XContentParser.Token.VALUE_STRING) {
-            return parser.text();
-        } else if (token == XContentParser.Token.VALUE_NUMBER) {
-            return parser.numberValue();
-        } else if (token == XContentParser.Token.VALUE_BOOLEAN) {
-            return parser.booleanValue();
-        } else if (token == XContentParser.Token.START_OBJECT) {
-            return readMap(parser, mapFactory);
-        } else if (token == XContentParser.Token.START_ARRAY) {
-            return readList(parser, mapFactory);
-        } else if (token == XContentParser.Token.VALUE_EMBEDDED_OBJECT) {
-            return parser.binaryValue();
+    static Object readValue(XContentParser parser, MapFactory<Object> mapFactory) throws IOException {
+        switch (parser.currentToken()) {
+            case VALUE_NULL: return null;
+            case VALUE_STRING: return parser.text();
+            case VALUE_NUMBER: return parser.numberValue();
+            case VALUE_BOOLEAN: return parser.booleanValue();
+            case START_OBJECT: return readMap(parser, mapFactory);
+            case START_ARRAY: return readList(parser, mapFactory);
+            case VALUE_EMBEDDED_OBJECT: return parser.binaryValue();
```

Comment:
can you use `default` here to `return null`?

Reply:
Done.